From oneukum@suse.de  Tue Jan 16 17:14:44 2007
From: Oliver Neukum <oneukum@suse.de>
Date: Sat, 13 Jan 2007 07:32:27 +0100
Subject: USB: race fixes for usb-serial, step 3
To: gregkh@suse.de, linux-usb-devel@lists.sourceforge.net, "J" <jhnlmn@yahoo.com>
Message-ID: <200701130732.27422.oneukum@suse.de>
Content-Disposition: inline

- fix an error code returned if a device has been disconnected

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/usb-serial.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/serial/usb-serial.c
+++ gregkh-2.6/drivers/usb/serial/usb-serial.c
@@ -281,7 +281,7 @@ static void serial_close(struct tty_stru
 static int serial_write (struct tty_struct * tty, const unsigned char *buf, int count)
 {
 	struct usb_serial_port *port = tty->driver_data;
-	int retval = -EINVAL;
+	int retval = -ENODEV;
 
 	if (!port || port->serial->dev->state == USB_STATE_NOTATTACHED)
 		goto exit;
@@ -289,6 +289,7 @@ static int serial_write (struct tty_stru
 	dbg("%s - port %d, %d byte(s)", __FUNCTION__, port->number, count);
 
 	if (!port->open_count) {
+		retval = -EINVAL;
 		dbg("%s - port not opened", __FUNCTION__);
 		goto exit;
 	}
